src/SolverInterface/constraints.jl

```diff
@@ -0,0 +1,77 @@
+
+
+"""
+    ConstraintReference{T}
+
+A lightweight object used to reference constraints in a model.
+The parameter `T` is the type of set constraint referenced.
+"""
+struct ConstraintReference{T}
+    value::UInt64
+end
+
+"""
+    candelete(m::AbstractMathProgModel, ref::ConstraintReference)::Bool
+
+Return a `Bool` indicating whether this constraint can be removed from the model `m`.
+"""
+candelete(m::AbstractMathProgModel, ref::ConstraintReference) = throw(MethodError())
+
+"""
+    isvalid(m::AbstractMathProgModel, ref::ConstraintReference)::Bool
+
+Return a `Bool` indicating whether this reference is valid
+for an active constraint in the model `m`.
+"""
+isvalid(m::AbstractMathProgModel, ref::ConstraintReference) = throw(MethodError())
+
+"""
+    delete!(m::AbstractMathProgModel, ref::ConstraintReference)
+
+Delete the referenced constraint from the model.
+
+    delete!(m::AbstractMathProgModel, refs::Vector{ConstraintReference})
+
+Delete the referenced constraints in the vector `refs` from the model.
+"""
+Base.delete!(m::AbstractMathProgModel, ref::ConstraintReference) = throw(MethodError())
+Base.delete!(m::AbstractMathProgModel, refs::Vector{ConstraintReference}) = throw(MethodError())
```

Comment:
The fallback should probably be
`Base.delete!(m::AbstractMathProgModel, refs::Vector{ConstraintReference}) = delete!.(m, refs)`